hist/hist/inc/TKDE.h

```diff
@@ -42,46 +46,63 @@ class TKDE : public TNamed  {
       kTotalKernels // Internal use only for member initialization
    };
 
-   enum EIteration { // KDE fitting option
+   /// Iteration types. They can be set using SetIteration()
+   enum EIteration {
       kAdaptive,
       kFixed
    };
 
-   enum EMirror { // Data "mirroring" option to address the probability "spill out" boundary effect
+   /// Data "mirroring" option to address the probability "spill out" boundary effect
+   /// They can be set using SetMirror()
+   enum EMirror {
       kNoMirror,
       kMirrorLeft,
       kMirrorRight,
       kMirrorBoth,
       kMirrorAsymLeft,
-      kMirrorAsymLeftRight,
+      kMirrorRightAsymLeft,
       kMirrorAsymRight,
       kMirrorLeftAsymRight,
       kMirrorAsymBoth
    };
 
-   enum EBinning{ // Data binning option
+   /// Data binning option.
+   /// They can be set using SetBinning()
+   enum EBinning{
       kUnbinned,
       kRelaxedBinning, // The algorithm is allowed to use binning if the data is large enough
       kForcedBinning
    };
 
-   
-   TKDE();                    // defaul constructor used only by I/O 
-
+   ///  default constructor used only by I/O
+   TKDE();
+
+   /// Constructor for unweighted data
+   /// Varius option for TKDE can be passed in the option string as below.
+   /// Note that min and max will define the plotting range but will not restrict the data in the unbinned case
+   /// Instead when use binning, only the data in the range will be considered.
+   /// Note also, that when some data exists outside the range, one should not use the mirror option with unbinned.
+   /// Adaptive will be soon very slow especially for Nevents > 10000.
+   /// For this reason, by default for Nevents >=10000, the data are automatically binned  in
+   /// nbins=Min(10000,Nevents/10)
+   /// In case of ForceBinning option the default number of bins is 1000
    TKDE(UInt_t events, const Double_t* data, Double_t xMin = 0.0, Double_t xMax = 0.0, const Option_t* option =
                  "KernelType:Gaussian;Iteration:Adaptive;Mirror:noMirror;Binning:RelaxedBinning", Double_t rho = 1.0) {
       Instantiate( nullptr,  events, data, nullptr, xMin, xMax, option, rho);
    }
 
+   /// Constructor for weighted data
    TKDE(UInt_t events, const Double_t* data, const Double_t* dataWeight, Double_t xMin = 0.0, Double_t xMax = 0.0, const Option_t* option =
         "KernelType:Gaussian;Iteration:Adaptive;Mirror:noMirror;Binning:RelaxedBinning", Double_t rho = 1.0) {
       Instantiate( nullptr,  events, data, dataWeight, xMin, xMax, option, rho);
    }
 
+   /// Constructor for unweweighted data and a user defined kernel function
```

Comment:
There is a typo: `unweweighted`

Reply:
By the way, is the documentation automatically checked for typos at some point? Just to know if it's actually necessary to sporadically pay attention to spelling in PR reviews.

Reply:
Thank you for checking the spelling. Unfortunately the documentation does not check for this automatically